app/vmselect/prometheus/prometheus.go

```diff
@@ -514,8 +514,12 @@ func LabelValuesHandler(qt *querytracer.Tracer, startTime time.Time, at *auth.To
 	if err != nil {
 		return err
 	}
+	minLimit := *maxUniqueTimeseries
+	if limit > 0 && limit < *maxUniqueTimeseries {
+		minLimit = limit
+	}
 	denyPartialResponse := searchutils.GetDenyPartialResponse(r)
-	sq := storage.NewSearchQuery(at.AccountID, at.ProjectID, cp.start, cp.end, cp.filterss, *maxUniqueTimeseries)
+	sq := storage.NewSearchQuery(at.AccountID, at.ProjectID, cp.start, cp.end, cp.filterss, minLimit)
```

Comment:
This limit may work incorrectly, since multiple time series may have the same `label="value"` pair. For example, if `limit=2` is passed to `/api/v1/label/labelName/values`, then it is expected that the query returns up to 2 values for the given `labelName`. If more than one time series has `labelName="foo"` label, then the new logic will return only the `foo` label value, while ignoring other possible label values from the remaining time series. So these limits work  independently.

Reply:
thanks，got it